From: Peter Dey <debian@realmtech.net>
Date: Wed, 5 Aug 2026 16:06:32 +0200
Subject: Include <pty.h> when openpty() is available so glibc builds succeed

Forwarded: https://savannah.gnu.org/bugs/index.php?68134
Last-Update: 2026-02-07
Comment: Debian enables -Werror=implicit-function-declaration in
 dpkg-buildflags, so glibc builds fail when configure sets HAVE_OPENPTY
 without providing a visible prototype.  Pull in <termios.h>/<pty.h> and
 declare the fallback prototype so the build succeeds.
---
 pty.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pty.c b/pty.c
index 5139403..99d8f4c 100644
--- a/pty.c
+++ b/pty.c
@@ -31,10 +31,11 @@
 #include "pty.h"
 
 #include <sys/ioctl.h>
+#include <termios.h>
 
 #include "screen.h"
 
-#if defined(HAVE_PTY_H)
+#if defined(HAVE_PTY_H) || defined(HAVE_OPENPTY)
 # include <pty.h>
 #endif
 #if defined(HAVE_UTIL_H)
@@ -44,6 +45,10 @@
 # include <libutil.h>
 #endif
 
+extern int openpty (int *__amaster, int *__aslave, char *__name,
+                    const struct termios *__termp,
+                    const struct winsize *__winp);
+
 int pty_preopen = 0;
 
 /***************************************************************/
